Source/WebCore/loader/LinkLoader.cpp

```diff
@@ -276,6 +281,9 @@ std::unique_ptr<LinkPreloadResourceClient> LinkLoader::preloadIfNeeded(const Lin
 
     if (cachedLinkResource && loader)
         return createLinkPreloadResourceClient(*cachedLinkResource, *loader, document);
+
+    if (loader)
+        loader->triggerError();
```

Comment:
What this patch seems to do so is to handle the case of synchronous failures.
It seems async failures would be reported. Maybe we can check this and beef up the commit log.

Reply:
Thanks for taking a look!

Yes I think that's right. Asynchronous failures are handled via the appropriate LinkPreloadResourceClient if I understand correctly. This handles the case when we don't even schedule the load with the Network Process such as when blocked by CSP.

The commit log could definitely use some improvement, I'll better explain myself in that.